// openai@5.12.2/src/lib/ChatCompletionRunner.ts
import {
  type ChatCompletionMessageParam,
  type ChatCompletionCreateParamsNonStreaming,
} from '.pnpm/openai@5.12.2/node_modules/openai/src/resources/chat/completions';
import { type BaseFunctionsArgs, RunnableTools } from '.pnpm/openai@5.12.2/node_modules/openai/src/lib/RunnableFunction';
import {
  AbstractChatCompletionRunner,
  AbstractChatCompletionRunnerEvents,
  RunnerOptions,
} from '.pnpm/openai@5.12.2/node_modules/openai/src/lib/AbstractChatCompletionRunner';
import { isAssistantMessage } from '.pnpm/openai@5.12.2/node_modules/openai/src/lib/chatCompletionUtils';
import OpenAI from '.pnpm/openai@5.12.2/node_modules/openai/src';
import { AutoParseableTool } from '.pnpm/openai@5.12.2/node_modules/openai/src/lib/parser';

export interface ChatCompletionRunnerEvents extends AbstractChatCompletionRunnerEvents {
  content: (content: string) => void;
}

export type ChatCompletionToolRunnerParams<FunctionsArgs extends BaseFunctionsArgs> = Omit<
  ChatCompletionCreateParamsNonStreaming,
  'tools'
> & {
  tools: RunnableTools<FunctionsArgs> | AutoParseableTool<any, true>[];
};

export class ChatCompletionRunner<ParsedT = null> extends AbstractChatCompletionRunner<
  ChatCompletionRunnerEvents,
  ParsedT
> {
  static runTools<ParsedT>(
    client: OpenAI,
    params: ChatCompletionToolRunnerParams<any[]>,
    options?: RunnerOptions,
  ): ChatCompletionRunner<ParsedT> {
    const runner = new ChatCompletionRunner<ParsedT>();
    const opts = {
      ...options,
      headers: { ...options?.headers, 'X-Stainless-Helper-Method': 'runTools' },
    };
    runner._run(() => runner._runTools(client, params, opts));
    return runner;
  }

  override _addMessage(
    this: ChatCompletionRunner<ParsedT>,
    message: ChatCompletionMessageParam,
    emit: boolean = true,
  ) {
    super._addMessage(message, emit);
    if (isAssistantMessage(message) && message.content) {
      this._emit('content', message.content as string);
    }
  }
}
